### PR TITLE
[WIP]add examples using Nano HPO to tune the hyperparameters in tensorflow training

### PR DESCRIPTION
**Pull request by Chaselzxy**
## Description

Add 5 examples using Nano HPO to tune the hyperparameters in tensorflow training.

### 1. Why the change?

Divide the tutorial of how to use Nano HPO to tune the hyperparameters in tensorflow training into several small examples.

### 3. Summary of the change 

 model arch search - using sequential API
 model arch search - using functional API
 model arch search - by subclassing keras.Model
 learning rate search
 batch size search

### 4. How to test?
test in tutorial

### CODE REVIEW

**Comment by shane-huang** [comment]
Each example should be a smallest runnable program which demonstrate just a single functionality. For example, for batch size search, you should only specify batch size for search (now you still have model arch search, i.e. search spaces in model build).

File: python/nano/tutorial/batch size search.ipynb [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "SNmZHFMMBfUr"
+   },
+   "outputs": [],
+   "source": [
+    "inputs = Input(shape=(28,28,1))\n",
+    "x = Conv2D(\n",
+    "    filters=space.Categorical(32, 64),\n",
```
Comment (shane-huang) [inline python/nano/tutorial/batch size search.ipynb:167]: we only need batch size search so remove the spaces in model layers.

File: python/nano/tutorial/learning rate search.ipynb [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    "        x = self.conv1(inputs)\n",
+    "        x = self.pool1(x)\n",
+    "        x = self.drop1(x)\n",
+    "        x = self.flat(x)\n",
+    "        x = self.dense1(x)\n",
+    "        x = self.drop3(x)\n",
+    "        x = self.dense2(x)\n",
+    "        return x\n",
+    "model = MyModel(\n",
+    "    filters=hpo.space.Categorical(32, 64),\n",
```
Comment (shane-huang) [inline python/nano/tutorial/learning rate search.ipynb:192]: remove the search space in layers. Just use a normal keras model which don't have any search spaces.

**Comment by shane-huang** [comment]
Replace spaces with “_” in example file names.

**Comment by shane-huang** [comment]
- turn these ipynb examples into .py
